ray_julia_jll/src/wrappers/any.jl

```diff
@@ -167,8 +168,26 @@ FromNil(::Type{ObjectID}) = ObjectIDFromNil()
 
 ObjectID(str::AbstractString) = FromHex(ObjectID, str)
 
-Base.show(io::IO, x::ObjectID) = show(io, "ObjectID(\"$(Hex(x))\")")
-Base.:(==)(a::ObjectID, b::ObjectID) = Hex(a) == Hex(b)
+Base.show(io::IO, x::ObjectID) = write(io, "ObjectID(\"", Hex(x), "\")")
+
+# cannot believe I'm doing this...
+# 
+# Because ObjectID is a CxxWrap-defined type, it has two subtypes:
+# `ObjectIDAllocated` and `ObjectIDDereferenced`.  The first is returned when we
+# construct directly or return by value, the second when you pull a ref out of
+# say `std::vector<ObjectID>`.
+#
+# ObjectID is abstract, so the normal method definition:
+# 
+# Base.:(==)(a::ObjectID, b::ObjectID) = Hex(a) == Hex(b)
+#
+# is shadowed by more specific fallbacks defined by CxxWrap.
+for Ta in subtypes(ObjectID)
+    for Tb in subtypes(ObjectID)
+        @eval Base.:(==)(a::$Ta, b::$Tb) = Hex(a) == Hex(b)
+    end
+end
```

Comment:
Aren't we invalidating some CxxWrap code by doing this? Seems like it's better just to avoid defining this and keeping the nice `==` in `ObjectRef`.

Also, does this work:
```julia
Base.:(==)(a::T1, b::T2) where T1 <: ObjectID, T2 <: ObjectID = Hex(a) == Hex(b)
```

Reply:
that's equivalent to `Base.:(==)(a::ObjectID, b::ObjectID)` right?

I'm not sure if we're invalidating CxxWrap-generated code, any more than we would be defining the original method.